From 6cb92b523bd60bd7718df134cc5b1eff51cf42e5 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Fri, 28 Jul 2023 22:11:45 +0300
Subject: [PATCH] Documentation: theme: Fix compilation with Sphinx >= 7.0.0

Sphinx 7.0.0 has dropped support for the deprecated 'style' variable
(https://github.com/sphinx-doc/sphinx/pull/11381). This breaks
compilation of the documentation:

  /usr/bin/sphinx-build -D release=v0.1.0+16-eed6a079 -q -W -b html Documentation Documentation/html

  Theme error:
  An error happened in rendering the page api-html/index.
  Reason: UndefinedError("'style' is undefined")

The recommended replacement is 'styles[-1]'. However, this resolves to
'_static/css/theme.css', which is part of the 'css_styles' variable, and
results in the stylesheet being included twice. To avoid that and fix
the compilation error, simply drop the first reference.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Tested-by: Kieran Bingham <kieran.bingham@ideasonboard.com>
Reviewed-by: Kieran Bingham <kieran.bingham@ideasonboard.com>
---
 Documentation/theme/layout.html | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Documentation/theme/layout.html b/Documentation/theme/layout.html
index fcc6d221..4fffefab 100644
--- a/Documentation/theme/layout.html
+++ b/Documentation/theme/layout.html
@@ -33,11 +33,6 @@ SPDX-License-Identifier: CC-BY-SA-4.0
 
   {% endif %}
 
-  {# RTD hosts this file, so just load on non RTD builds #}
-  {% if not READTHEDOCS %}
-    <link rel="stylesheet" href="{{ pathto('_static/' + style, 1) }}" type="text/css" />
-  {% endif %}
-
   {% for cssfile in css_files %}
     <link rel="stylesheet" href="{{ pathto(cssfile, 1) }}" type="text/css" />
   {% endfor %}
-- 
2.39.1

